plugin/pkg/scheduler/scheduler.go

```diff
@@ -233,6 +234,76 @@ func (sched *Scheduler) preempt(preemptor *v1.Pod, scheduleErr error) (string, e
 	return node.Name, err
 }
 
+// Assume volumes and then asynchronously bind volumes if required.
+// This function modifies assumed if volume binding is required.
+func (sched *Scheduler) assumeAndBindVolumes(assumed *v1.Pod, host string) error {
+	if utilfeature.DefaultFeatureGate.Enabled(features.VolumeScheduling) {
+		// TODO: can we distingush between fully bound and waiting for binding to complete?
+		// If it's waiting for binding to complete, then we can return early here instead of
+		// spawning a go routine for bindVolumes to do nothing...
+		bindingRequired, err := sched.config.VolumeBinder.AssumePodVolumes(assumed, host)
+		if err != nil {
+			sched.config.Error(assumed, err)
+			sched.config.Recorder.Eventf(assumed, v1.EventTypeWarning, "FailedScheduling", "AssumePod failed: %v", err)
+			sched.config.PodConditionUpdater.Update(assumed, &v1.PodCondition{
+				Type:    v1.PodScheduled,
+				Status:  v1.ConditionFalse,
+				Reason:  "SchedulerError",
+				Message: err.Error(),
+			})
+			return err
+		}
+		if bindingRequired {
+			if sched.config.Ecache != nil {
+				invalidPredicates := sets.NewString("NoVolumeBindConflict", "NoVolumeZoneConflict")
```

Comment:
I think volume zone predicate doesn't need to be invalidated because it's not handling pvcs with the new wait binding mode.